### DISCUSSION TITLE
Loading texture for obj

### DISCUSSION THREAD

**Discussion by Theodliu** [Kitware/trame#352] [Q&A] [answered]
Hi!
I am trying to load a .mtl file for .obj to load texture for obj. Can trame load .mtl files or jpg, dds files for the texture?

Thanks again.

**Comment by jourdain** [comment]
Is the question related to local rendering or [server-side VTK](https://vtk.org/doc/nightly/html/classvtkOBJImporter.html#details)?

**Comment by jourdain** [comment]
The VtkLocalView may be missing a couple of things, but technically vtk.js can render those. It will just be a matter of fixing the serializer.

**Reply by Theodliu** [reply]
Sorry, I was confused about the vtkGLTFImporter(). Can this function load the textures? It seems some texture will missing because it doesn't support several extensions:
glTF extension KHR_texture_transform is used in this model, but not supported by this loader. The extension will be ignored.
And another warning is that: 
vtkGLTFImporter.cxx:198   WARN| (nullptr): Using multiple texture coordinates for the same model is not supported.

Also, I find that there is a function called vtkGLTFReader(), but I don't know how to use this function. So what's the difference between the vtkGLTFReader() and vtkGLTFImporter()?

Thanks again.

**Reply by Theodliu** [reply]
Extensions like KHR_texture_transform, KHR_materials_pbrSpecularGlossiness are ignored.

**Reply by Theodliu** [reply]
And I found there are several methods in vtkGLTFImporter(). But EnableAnimation() can't show the animation. Do I miss anything?

**Reply by jourdain** [reply, marked as answer]
It seems to me that you have a VTK question. https://discourse.vtk.org/
